src/Spryker/Zed/ProductStorage/Persistence/ProductStorageQueryContainer.php

```diff
@@ -39,7 +39,6 @@ public function queryProductAbstractByIds(array $productAbstractIds)
             ->joinWithLocale()
             ->joinWithSpyProductAbstract()
             ->useSpyProductAbstractQuery()
-                ->joinWithSpyProduct()
```

Comment:
QueryContainer methods are public API. Meaning, that changes that change its behavior are considered as a BC break.

In this case I would suggest deprecate this method or add a new optional param, that is true by default, that shows -  should we add join or not

Reply:
Done via deprecation